### ISSUE TITLE
`prefer-spread`: Add prefer spread over `.slice()/.concat()`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#737]
Currently, [`prefer-spread`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-spread.md) only check `Array.from()`, how about add prefer spread over `.slice()/.concat()`?

# Fail

```js
const foo = array.slice();
```

```js
const foo = array.concat(item);
```

```js
const foo = array.concat(anotherArray);
```

# Pass

```js
const foo = [...array];
```

```js
const foo = array.slice(1);
```

The `concat()` case, we can't know `item` is array or not. 

Maybe with suggestions? 
Suggestion 1: `[...array, itemOrArray]`
Suggestion 2: `[...array, ...itemOrArray]`

Another problem is `TypedArray#slice()` and `String#concat()`. `String#slice` should not a problem. Nobody use `string.slice()`(no arguments).

---
Any other method people use to clone an array?

**Comment by sindresorhus** [comment]
> The concat() case, we can't know item is array or not.

Maybe, if the user is using the TypeScript parser (https://github.com/sindresorhus/eslint-plugin-unicorn/issues/347), we could handle it?

And if not, just use suggestions.

**Comment by sindresorhus** [comment]
> Any other method people use to clone an array?

`Array.from([1, 2])` and `Array.of([1, 2])`.

**Comment by fisker** [comment]
@sindresorhus I mean add to our existing rule [`prefer-spread`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-spread.md), not new rule.

**Comment by fisker** [comment]
I feel bad that I can't distinguish `Array` and `TypedArray`, so I can't make this rule work on `Array#slice()`

Let's close this?

**Comment by sindresorhus** [comment]
@fisker Maybe we can make it opt-in? I have personally never used or seen `.slice()` used on TypedArray. For me, it's more valuable to catch `Array#slice()`. `// eslint-disable` are good for edge-cases like this.

**Comment by coreyfarrell** [comment]
Are you open to an option to ignore `[].concat(value)`?  I use this pattern to normalize arguments where a single element or an array of elements is accepted.  I know I could use `Array.isArray(value) ? [...value] : [value]` but IMO this is less readable.

**Comment by sindresorhus** [comment]
@coreyfarrell You can just use `[value].flat()` for that (we will eventually enforce that https://github.com/sindresorhus/eslint-plugin-unicorn/issues/975).

**Comment by coreyfarrell** [comment]
That is a reasonable replacement, thanks!
